orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/services/LogoutService.java

```diff
@@ -177,21 +192,41 @@ public APIGatewayProxyResponseEvent handleAccountInterventionLogout(
 
         cloudwatchMetricsService.incrementLogout(Optional.of(clientId), Optional.of(intervention));
         return generateLogoutResponse(
-                redirectURI, Optional.empty(), auditUser, Optional.of(clientId), Optional.empty());
+                redirectURI,
+                LogoutReason.INTERVENTION,
+                auditUser,
+                Optional.of(clientId),
+                Optional.empty());
     }
 
     private void sendAuditEvent(
-            TxmaAuditUser auditUser, Optional<String> clientId, Optional<String> rpPairwiseId) {
+            TxmaAuditUser auditUser,
+            LogoutReason logoutReason,
+            Optional<String> clientId,
+            Optional<String> rpPairwiseId) {
         String auditClientId = clientId.orElse(AuditService.UNKNOWN);
-        var metadata =
-                rpPairwiseId
-                        .map(i -> new AuditService.MetadataPair[] {pair("rpPairwiseId", i)})
-                        .orElse(new AuditService.MetadataPair[] {});
-        auditService.submitAuditEvent(LOG_OUT_SUCCESS, auditClientId, auditUser, metadata);
+        var metadata = new LinkedList<AuditService.MetadataPair>();
+        metadata.add(pair(LOGOUT_REASON, logoutReason.getValue()));
+        rpPairwiseId.ifPresent(i -> metadata.add(pair("rpPairwiseId", i)));
+        auditService.submitAuditEvent(
+                LOG_OUT_SUCCESS,
+                auditClientId,
+                auditUser,
+                metadata.toArray(new AuditService.MetadataPair[0]));
```

Comment:
What's the difference between this and 
```suggestion
                metadata.toArray(AuditService.MetadataPair[]::new));
```
?

Reply:
Agreed, the second one is better